src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -0,0 +1,440 @@
+            byte[] fixedLenByteArrayVal =
+                getFixedLenByteArrayValue(value, primitiveType, stats, insertRowsCurrIndex);
+            value = fixedLenByteArrayVal;
+            estimatedParquetSize +=
+                ParquetBufferValue.BYTE_ARRAY_LENGTH_ENCODING_BYTE_LEN
+                    + fixedLenByteArrayVal.length;
+            break;
+          default:
+            throw new SFException(
+                ErrorCode.UNKNOWN_DATA_TYPE,
+                type.getLogicalTypeAnnotation(),
+                primitiveType.getPrimitiveTypeName());
+        }
+      } else {
+        return getGroupValue(
+            value, type.asGroupType(), stats, defaultTimezone, insertRowsCurrIndex);
+      }
+    }
+
+    if (value == null) {
+      if (!type.isRepetition(Repetition.REQUIRED)) {
+        throw new SFException(
+            ErrorCode.INVALID_FORMAT_ROW, type.getName(), "Passed null to non nullable field");
+      }
+      stats.incCurrentNullCount();
+    }
+
+    return new ParquetBufferValue(value, estimatedParquetSize);
+  }
+
+  /**
+   * Parses an int32 value based on Parquet logical type.
+   *
+   * @param value column value provided by user in a row
+   * @param type Parquet column type
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return parsed int32 value
+   */
+  private static int getInt32Value(
+      Object value, PrimitiveType type, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return DataValidationUtil.validateAndParseIcebergNumber(
+          type.getName(), value, insertRowsCurrIndex, Integer.class);
+    }
+    if (logicalTypeAnnotation instanceof DecimalLogicalTypeAnnotation) {
+      return getDecimalValue(value, type, insertRowsCurrIndex).unscaledValue().intValue();
+    }
+    if (logicalTypeAnnotation == LogicalTypeAnnotation.dateType()) {
+      return DataValidationUtil.validateAndParseDate(type.getName(), value, insertRowsCurrIndex);
+    }
+    throw new SFException(
+        ErrorCode.UNKNOWN_DATA_TYPE, logicalTypeAnnotation, type.getPrimitiveTypeName());
+  }
+
+  /**
+   * Parses an int64 value based on Parquet logical type.
+   *
+   * @param value column value provided by user in a row
+   * @param type Parquet column type
+   * @param insertRowsCurrIndex Used for logging the row of index given in insertRows API
+   * @return parsed int64 value
+   */
+  private static long getInt64Value(
+      Object value, PrimitiveType type, ZoneId defaultTimezone, final long insertRowsCurrIndex) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation == null) {
+      return DataValidationUtil.validateAndParseIcebergNumber(
+          type.getName(), value, insertRowsCurrIndex, Long.class);
+    }
+    if (logicalTypeAnnotation instanceof DecimalLogicalTypeAnnotation) {
```

Comment:
nit: line 154 uses a different style for a similar kind of check: `logicalTypeAnnotation == LogicalTypeAnnotation.dateType`
Lets do one style in all places (instanceof lgtm) ?

Reply:
Changed to instanceof.